backend/general_routes.py

```diff
@@ -117,8 +134,220 @@ def update_schema(data: schemas.SchemaUpdateSchema, id_or_slug: Union[int, str],
+    return auth_crud.get_users(db=db)
+
+
+@router.get('/users/{username}/memberships', tags=["Auth"],
+            summary="Groups the user is a member of", response_model=List[GroupSchema])
+def get_user_groups(username: str, db: Session = Depends(get_db),
+                    auser: User = Depends(authenticated_user)) -> List[GroupSchema]:
+    user = auth_crud.get_user(username=username, db=db)
+    return auth_crud.get_user_groups(user=user, db=db)
+
+
+@router.patch('/users/{username}', tags=["Auth"], summary="Activate user", responses={
+    200: {"description": "User activated"},
+    208: {"description": "User is already active"},
+    404: {"description": "No such user exists"}
+})
+def activate_user(username: str, response: Response, db: Session = Depends(get_db),
+                  auser: User = Depends(authorized_user(RequirePermission(permission=PermissionType.USER_MANAGEMENT)))) -> bool:
+    try:
+        changed = auth_crud.activate_user(username=username, db=db)
+        if not changed:
+            response.status_code = status.HTTP_208_ALREADY_REPORTED
+    except ValueError as error:
+        raise HTTPException(status.HTTP_400_BAD_REQUEST, str(error))
+    except exceptions.MissingUserException as error:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, str(error))
+    else:
+        return changed
+
+
+@router.delete('/users/{username}', tags=["Auth"], summary="Deactivate user", responses={
+    200: {"description": "User deactivated"},
+    208: {"description": "User is already deactivated"},
+    404: {"description": "No such user exists"}
+})
+def deactivate_user(username: str, response: Response, db: Session = Depends(get_db),
+                    auser: User = Depends(authorized_user(RequirePermission(permission=PermissionType.USER_MANAGEMENT)))) -> bool:
+    try:
+        changed = auth_crud.deactivate_user(username=username, db=db)
+        if not changed:
+            response.status_code = status.HTTP_208_ALREADY_REPORTED
+    except ValueError as error:
+        raise HTTPException(status.HTTP_400_BAD_REQUEST, str(error))
+    except exceptions.MissingUserException as error:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, str(error))
+    else:
+        return changed
+
+
+@router.get('/permissions', tags=["Auth"], summary="Get list of granted permissions")
+def get_permissions(recipient_type: Optional[RecipientType] = Query(None, description="Type (User or Group) of permission recipient"),
+                    recipient_id: Optional[int] = Query(None, description="ID of permission recipient"),
+                    obj_type: Optional[PermissionTargetType] = Query(None, description="Type of target object"),
+                    obj_id: Optional[int] = Query(None, description="ID of target object"),
+                    db: Session = Depends(get_db),
+                    user: User = Depends(authenticated_user)):
```

Comment:
Allowing any authenticated user to fetch a listing of alll granted permissions might be considered a security risk :thinking: 

Also, the `USER_MANAGEMENT` permission might need some refinement, e.g. on the schema/entity level.

IMHO, this should be best addressed in a separate issue.

Reply:
#39